tests/jeos/image_info.pm

```diff
@@ -0,0 +1,97 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: This module is to collect information about JeOS/Minimal-VM
+#          information, like image size, directory sizes, package list,
+#          etc. and push it to an InfluxDB to be displayed in Grafana.
+#          This module is not intended to be fatal if something fails.
+# Maintainer: QA-C <qa-c@suse.de>
+
+use Mojo::Base qw(opensusebasetest);
+use testapi;
+use version_utils qw(is_sle is_opensuse);
+use mmapi qw(get_current_job_id);
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    my $influx_server = get_required_var('INFLUX_SERVER');
+    my $openqa_host = get_required_var('OPENQA_HOSTNAME');
+    my $download_url;
+    my $job_url;
+
+    my $host = $bmwqemu::vars{OPENQA_URL};
+    if ($openqa_host =~ /openqa1-opensuse/) {
+        $download_url = "http://openqa.opensuse.org/assets/hdd";
+        $job_url = 'https://openqa.opensuse.org/tests/' . get_current_job_id();
+    }
+    elsif ($openqa_host =~ /openqa.suse.de/) {
+        $download_url = "https://openqa.suse.de/assets/hdd";
+        $job_url = 'https://openqa.suse.de/tests/' . get_current_job_id();
+    }
+    else {
+        record_info("Skip", "This job is not running from OSD or O3.");
+        return;
+    }
+
+    # We need to get the image from ASSET_256 variable. Some jobs like OpenStack don't
+    # boot a JeOS image but a pre-installed jumphost. However, those special jobs have
+    # ASSET_256 pointing to the right JeOS image checksum.
+    my $asset = get_required_var('ASSET_256');
+    $asset =~ s/\.sha256//;    # to remove .sha256 sufix
+    my $asset_url = "$download_url/$asset";
+    # Check url exists
+    if (script_output("curl -I -k $asset_url") !~ /HTTP.*200/) {
+        record_soft_failure("Asset not found in $asset_url");
+        return;
+    }
+    (my $image = $asset) =~ s/\.xz//;    # in case the image is compressed
+    record_info('Image', $image);
+
+    # We shall skip this module for verification runs with special build numbers
+    if (get_required_var('BUILD') =~ /os-autoinst-distri-opensuse/) {
```

Comment:
I do verification runs using `openqa-clone-job`, so only `CASEDIR` and `_GROUP` are different. Something similar appears to be the case for automatic investigation jobs like https://openqa.opensuse.org/tests/2307416#settings. I suggest to check for `/` in `BUILD` or any non-default `CASEDIR`, but the latter might also have false positives in case an explicit version is given.

Or maybe just hardcode the `_GROUP` numbers for production? Not sure whether that is available in test modules at all.

Reply:
The `CASEDIR` condition alone is not sufficient. I occasionally clone jobs to my own instance and we don't want those test runs to contaminate the database.